packages/features/bookings/lib/checkIfUserIsAuthorizedToManageBooking.ts

```diff
@@ -0,0 +1,55 @@
+import { EventTypeRepository } from "@calcom/features/eventtypes/repositories/eventTypeRepository";
+import { MembershipRepository } from "@calcom/features/membership/repositories/MembershipRepository";
+import { PrismaOrgMembershipRepository } from "@calcom/features/membership/repositories/PrismaOrgMembershipRepository";
+import { prisma } from "@calcom/prisma";
+import { UserPermissionRole } from "@calcom/prisma/enums";
+
+type CheckIfUserIsAuthorizedToManageBookingParams = {
+  eventTypeId: number | null;
+  loggedInUserId: number;
+  teamId?: number | null;
+  bookingUserId: number | null;
+  userRole: string;
+};
+
+export const checkIfUserIsAuthorizedToManageBooking = async ({
+  eventTypeId,
+  loggedInUserId,
+  teamId,
+  bookingUserId,
+  userRole,
+}: CheckIfUserIsAuthorizedToManageBookingParams): Promise<boolean> => {
+  // System-wide admin
+  if (userRole === UserPermissionRole.ADMIN) return true;
+
+  // Organizer/owner of the booking
```

Comment:
I wonder if it's worth writing named variables instead of comments. Not blocking, but worth considering
```suggestion
  const isSystemWideAdmin = userRole === UserPermissionRole.ADMIN;
  if (isSystemWideAdmin) return true;
```
for example